packages/library-base/src/commonMain/kotlin/io/realm/kotlin/log/RealmLog.kt

```diff
@@ -0,0 +1,390 @@
+            field = value
+        }
+
+    // Lock preventing multiple threads modifying the list of loggers.
+    private val loggersMutex = SynchronizableObject()
+    // Reference to the currently installed system logger (if any)
+    // `internal` until we can remove the old log API
+    internal var systemLoggerInstalled: RealmLogger? = null
+    // Kotlin Multiplatform are currently lacking primitives like CopyOnWriteArrayList. We could
+    // use `io.realm.kotlin.internal.interop.SynchronizableObject`, but it would require locking
+    // when reporting a log statement which feel a bit heavy, so instead we have added locks around
+    // all modifications to this array (which are expected to be rare) and the `doLog` method must
+    // copy this reference before using it.
+    private var loggers: MutableList<RealmLogger> = mutableListOf()
+
+    init {
+        addDefaultSystemLogger()
+        RealmInterop.realm_set_log_callback(
+            level.toCoreLogLevel(),
+            object : LogCallback {
+                override fun log(logLevel: Short, message: String?) {
+                    // Create concatenated up front, since Core should already filter messages
+                    // not within the log range.
+                    val level: LogLevel = fromCoreLogLevel(CoreLogLevel.valueFromPriority(logLevel))
+                    doLog(
+                        level,
+                        null,
+                        if (message.isNullOrBlank()) { null } else { "[Core] $message" }
+                    )
+                }
+            }
+        )
+    }
+
+    /**
+     * Add a logger that will be notified on log events that are equal to or exceed the currently
+     * configured [level].
+     *
+     * @param logger logger to add.
+     */
+    public fun add(logger: RealmLogger) {
+        loggersMutex.withLock {
+            loggers.add(logger)
+        }
+    }
+
+    /**
+     * Removes the given logger if possible.
+     *
+     * @param logger logger that should be removed.
+     * @return `true` if the logger was removed, `false` if it wasn't registered.
+     */
+    public fun remove(logger: RealmLogger): Boolean {
+        loggersMutex.withLock {
+            val updatedLoggers = MutableList(loggers.size) { loggers[it] }
+            return updatedLoggers.remove(logger).also {
+                loggers = updatedLoggers
+            }
+        }
+    }
+
+    /**
+     * Removes all loggers. The default system logger will be removed as well unless
+     * [removeDefaultSystemLogger] is set to `false`. [addDefaultSystemLogger] can be used
+     * to add the default logger again if it was removed.
+     *
+     * @param removeDefaultSystemLogger whether or not to also remove the default system logger.
+     * @return `true` will be returned if one or more loggers were removed, `false` if no loggers were
+     * removed.
+     */
+    public fun removeAll(removeDefaultSystemLogger: Boolean = true): Boolean {
```

Comment:
A bit clumsy with this `removeDefaultSystemLogger`-argument. Maybe just make `removeAll` remove them all and do a `removeAllCustomLoggers()` for keeping the system logger.

Reply:
Hmm, yeah 🤔 , I guess I just wanted to avoid having two methods for this since we don't have these boolean methods elsewhere in the API, but I do agree the named parameters end up being a bit weird. 

My thinking was that `removeAll()` or `removeAll(true)` should remove everything at least.... which kinda destroys  `removeAll(keepSystemLogger = false)` even though that reads a bit better.

Reply:
We could also just `removeAll()` and then you would have to `addSystemDefaultLogger()` if you want to reinclude the system logger.

Reply:
That was the approach we took in Java, so yeah, maybe we should just do that here as well 👍